src/hotspot/cpu/x86/stubGenerator_x86_64.cpp

```diff
@@ -2572,6 +2631,75 @@ address StubGenerator::generate_base64_decodeBlock() {
     __ BIND(L_bruteForce);
   }   // End of if(avx512_vbmi)
 
+  if (VM_Version::supports_avx2()) {
+    Label L_tailProc, L_topLoop, L_enterLoop;
+    ////////////////////////////////////////////
+    //
+    //  **** Make sure inputs match ****
+    // size_t fast_avx2_base64_decode(char *out, const char *src, size_t srclen)
+    //
+    // Handle isURL / MIME?!?!  r12 is used for length calculation (from out)
+    //
+    // rbx is out, r12 is saved out, rdx is size, rsi is src
+    //
+    ////////////////////////////////////////////
+
+    // *************** NEEDS TO BE FIXED ************
+    __ cmpl(isURL, 0);
+    __ jcc(Assembler::notZero, L_tailProc);
+
+    __ cmpl(length, 44);
+    __ jcc(Assembler::belowEqual, L_tailProc);
+
+    __ cmpl(isMIME, 0);
+    __ jcc(Assembler::notEqual, L_tailProc);
+
+    // Set up constants
+    __ lea(r13, ExternalAddress(StubRoutines::x86::base64_AVX2_tables_addr()));
+    __ vpbroadcastq(xmm4, Address(r13, 0), Assembler::AVX_256bit);
+    __ vmovdqu(xmm11, Address(r13, 0x28));
+    __ vpbroadcastb(xmm10, Address(r13, 0), Assembler::AVX_256bit);
```

Comment:
Sorry in advance since I'm probably reading this wrong: the data that `r13` is pointing to appears to be a repeated byte pattern (`0x2f2f2f...`), does this mean this `vpbroadcastb` and the `vpbroadcastq` above end up filling up their respective registers with the exact same bits? If so, and since neither of them is mutated in the code below, then perhaps this can be simplified a bit.

Reply:
You're reading it correctly - this is redundant and could be handled differently, as the same value is being loaded into ymm4 and ymm10.  I don't think there will be any significant performance gain either way.  This was done in this manner to allow easier transition to URL acceleration when it is implemented, as URLs require handling '-' and '_' instead of '+' and '/' ('/' = 0x2f).